contracts/multisig/src/key.rs

```diff
@@ -284,10 +333,122 @@ mod test {
         )
         .unwrap();
 
-        let signature = Signature::try_from((KeyType::Ecdsa, test_data::signature())).unwrap();
-        let message = MsgToSign::try_from(test_data::message()).unwrap();
+        let signature =
+            Signature::try_from((KeyType::Ecdsa, ecdsa_test_data::signature())).unwrap();
+        let message = MsgToSign::try_from(ecdsa_test_data::message()).unwrap();
         let public_key = PublicKey::try_from((KeyType::Ecdsa, invalid_pub_key)).unwrap();
         let result = signature.verify(&message, &public_key).unwrap();
         assert_eq!(result, false);
     }
 }
+
+#[cfg(test)]
+mod ed25519_tests {
```

Comment:
I like the separation into multiple modules!

Reply:
Thank you!